Fixes #2302: Introduced C API for Text-to-speech pipeline and SpeechT5

Nandkishore-04 opened the pull request:
Fixes #2302 : C bindings for Text-to-speech API

Description
Introduces C API bindings for the Text-to-speech (TTS) pipeline and SpeechT5 support.

- Implements C wrappers for 'Text2SpeechPipeline' and 'SpeechGenerationConfig'
- Exposes 'num_generated_samples' in the PerfMetrics C API
- Adds functional C sample 'text2speech_sample_c.c' with CMake integration
- Updates 'test_text2speech.py' to verify C sample in CI

Fixes #2302 

## Checklist:
- [x] Tests have been updated or added to cover the new code
- [x] This patch fully addresses the ticket
- [x] I have made corresponding changes to the documentation
- [x] clang-format applied to all C++/Python files
- [x] 'pytest tests/ -v' passes locally

Signed-off-by: Nand Kishore <nkk01414@gmail.com>

Wovchena reviewed (approved):
Fix Lint and wait for review from @as-suvorov and @rkazants

rkazants commented on samples/c/speech_generation/text2speech/text2speech_sample_c.c:
```diff
@@ -0,0 +1,165 @@
+    free(data);
+
+    return tensor;
+}
```
please move these helper functions and macro into `text2speech_sample_utils.h` and `text2speech_sample_utils.c` files

rkazants commented on samples/c/speech_generation/text2speech/text2speech_sample_c.c:
```diff
@@ -0,0 +1,165 @@
+    CHECK_STATUS(ov_genai_text2speech_decoded_results_get_speech_at(results, 0, &speech_tensor));
+
+    void* waveform_data = NULL;
+    ov_tensor_data(speech_tensor, &waveform_data);
```
please include "openvino/c/ov_tensor.h" explicitly in this file

rkazants commented on src/c/src/text2speech_pipeline.cpp:
```diff
@@ -0,0 +1,214 @@
+
+#include "openvino/genai/c/text2speech_pipeline.h"
+
+#include <stdarg.h>
```
why do we need this include? I think - not needed. Please double-check.

rkazants reviewed (commented):
Looks good to me. Left some comments.

Nandkishore-04 commented:
@rkazants All requested changes have been implemented: refactored the sample utilities, cleaned up includes globally, and updated the copyright year to 2026.

freddychiu commented on src/c/include/openvino/genai/c/text2speech_pipeline.h:
```diff
@@ -0,0 +1,137 @@
+ * @param device Name of a device to load a model to.
+ * @param property_args_size How many properties args will be passed, each property contains 2 args: key and value.
+ * @param pipe A pointer to the newly created ov_genai_text2speech_pipeline.
+ * @param ... property parameter: Optional pack of pairs: <char* property_key, char* property_value> relevant only
```
Can we type & ingest opaque handle or define it such that a list pointer(s) is explicit based on property_args_size? Given C APIs provide ABI stability, this change would enable better scalability to build on top of this API.

This same feedback applies to all variadic functions (...)

rkazants replied:
Hi @freddychiu, I think vlm and whisper C APIs also suffer from this issue. I propose to fix this limitation in separate PR for all pipelines.

rkazants commented on src/c/src/whisper_pipeline.cpp:
```diff
@@ -1,11 +1,12 @@
 #include <filesystem>
+#include <memory>
+#include <string>
+#include <vector>
```
please revert these changes unrelated to your PR

Wovchena commented:
@as-suvorov, can it be merged?

Nandkishore-04 commented:
hey @Wovchena , Should all the CI checks pass before this can be merged? I see some ongoing failures in Lint and Build Archive.

Wovchena commented:
build_jenkins

Nandkishore-04 commented:
hey @Wovchena , I found that an undeclared identifier f32 in 'text2speech_sample_utils.c' was the reason for the CI build failures. I corrected this to the standard OpenVINO C API constant F32 and also improved error handling/return codes to ensure any future issues are correctly reported to the CI runner. It is now ready to review!

Wovchena commented on samples/c/speech_generation/text2speech_sample_utils.c:
```diff
@@ -1,12 +1,11 @@
-// Copyright (C) 2026 Intel Corporation
-// SPDX-License-Identifier: Apache-2.0
-//
-
```
Put it back

rkazants reviewed (changes requested):
Please revert changes unrelated to TTS pipeline (like vlm, etc.) and do not implement c_element_type_to_cpp. Just take a look it is used in other samples.

Nandkishore-04 commented:
@goyaladitya05 , thanks for reaching me out! sure, but let me look at it once again and try to fix it. I'll reach out to you if I need any assistance.

Nandkishore-04 commented:
hey, @Wovchena the code is ready for review!

Wovchena reviewed (commented):
2026-02-05T11:31:50.8506549Z /home/runner/_work/openvino.genai/openvino.genai/src/src/c/src/text2speech_pipeline.cpp: In function 'ov_element_type_e cpp_element_type_to_c(const ov::element::Type&)':
2026-02-05T11:31:50.8512911Z /home/runner/_work/openvino.genai/openvino.genai/src/src/c/src/text2speech_pipeline.cpp:36:64: error: 'undefined' is not a member of 'ov::element::Type_t'
2026-02-05T11:31:50.8519173Z    36 |     return static_cast<ov_element_type_e>(ov::element::Type_t::undefined);
2026-02-05T11:31:50.8522920Z       |                                                                ^~~~~~~~~
2026-02-05T11:31:50.8536698Z gmake[2]: *** [src/c/CMakeFiles/openvino_genai_c.dir/build.make:163: src/c/CMakeFiles/openvino_genai_c.dir/src/text2speech_pipeline.cpp.o] Error 1
2026-02-05T11:31:50.8540968Z gmake[2]: *** Waiting for unfinished jobs....
2026-02-05T11:31:54.8597818Z gmake[2]: Leaving directory '/home/runner/_work/openvino.genai/openvino.genai/build'
2026-02-05T11:31:54.8598810Z gmake[1]: *** [CMakeFiles/Makefile2:2727: src/c/CMakeFiles/openvino_genai_c.dir/all] Error 2
2026-02-05T11:31:54.8599635Z gmake[1]: Leaving directory '/home/runner/_work/openvino.genai/openvino.genai/build'
2026-02-05T11:31:54.8600751Z gmake: *** [Makefile:156: all] Error 2

Nandkishore-04 committed:
Fixes #2302: Added C API bindings for Text2SpeechPipeline

Nandkishore-04 commented:
hey @Wovchena @rkazants , I have resolved all the issues, the code is ready for review.

Nandkishore-04 committed:
Merge branch 'master' into master

Nandkishore-04 committed:
Merge branch 'master' into master
